src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -178,11 +177,27 @@ private void WriteHeaders(HttpRequestMessage request)
                 WriteHeader(ref state, ":authority", authority);
                 WriteHeader(ref state, ":path", request.RequestUri.GetComponents(UriComponents.PathAndQuery | UriComponents.Fragment, UriFormat.UriEscaped));
 
+                // Determine cookies to send.
+                string cookiesFromContainer = null;
```

Comment:
I think this code should be further down in the 'if' block that does the `WriteHeader()` method using the 'cookiesFromContainer' parameter.  That keeps nearest the code that actually uses the 'cookiesFromContainer' variable.  And then you don't need to change it from empty string to null.

And then that code becomes simpler:

```c#
if (_connection.UseCookies)
{
    string cookiesFromContainer = _connection.CookieContainer.GetCookieHeader(request.RequestUri);
    if (cookiesFromContainer != string.Empty)
    {
        WriteHeader(ref state, HttpKnownHeaderNames.Cookie, cookiesFromContainer);
    }
}
```

Reply:
I forgot to do a clean up after removing grouping Cookie related header fields.

Will fix.